yb-voyager/src/query/queryparser/traversal_test.go

```diff
@@ -237,9 +239,117 @@ func TestObjectCollector2(t *testing.T) {
+		},
+		{
+			Sql:             `SELECT * from s1.employees`,
+			ExpectedObjects: []string{"s1.employees"},
+			ExpectedSchemas: []string{"s1"},
+		},
+		{
+			Sql:             `SELECT * from s2.employees where salary > (Select salary from s3.employees)`,
+			ExpectedObjects: []string{"s3.employees", "s2.employees"},
+			ExpectedSchemas: []string{"s3", "s2"},
+		},
+		{
+			Sql:             `SELECT c.name, SUM(o.amount) AS total_spent FROM sales.customers c JOIN finance.orders o ON c.id = o.customer_id GROUP BY c.name HAVING SUM(o.amount) > 1000`,
+			ExpectedObjects: []string{"sales.customers", "finance.orders"},
+			ExpectedSchemas: []string{"sales", "finance"},
+		},
+		{
+			Sql:             `SELECT name FROM hr.employees WHERE department_id IN (SELECT id FROM public.departments WHERE location_id IN (SELECT id FROM eng.locations WHERE country = 'USA'))`,
+			ExpectedObjects: []string{"hr.employees", "public.departments", "eng.locations"},
+			ExpectedSchemas: []string{"hr", "public", "eng"},
+		},
+		{
+			Sql:             `SELECT name FROM sales.customers UNION SELECT name FROM marketing.customers;`,
+			ExpectedObjects: []string{"sales.customers", "marketing.customers"},
+			ExpectedSchemas: []string{"sales", "marketing"},
+		},
+		{
+			Sql: `CREATE VIEW analytics.top_customers AS
+					SELECT user_id, COUNT(*) as order_count	FROM public.orders GROUP BY user_id HAVING COUNT(*) > 10;`,
+			ExpectedObjects: []string{"analytics.top_customers", "public.orders"},
+			ExpectedSchemas: []string{"analytics", "public"},
+		},
+		{
+			Sql: `WITH user_orders AS (
+					SELECT u.id, o.id as order_id
+					FROM users u
+					JOIN orders o ON u.id = o.user_id
+					WHERE o.amount > 100
+				), order_items AS (
+					SELECT o.order_id, i.product_id
+					FROM order_items i
+					JOIN user_orders o ON i.order_id = o.order_id
+				)
+				SELECT p.name, COUNT(oi.product_id) FROM products p
+				JOIN order_items oi ON p.id = oi.product_id GROUP BY p.name;`,
+			ExpectedObjects: []string{"users", "orders", "order_items", "user_orders", "products"},
+			ExpectedSchemas: []string{""},
+		},
+		{
+			Sql: `UPDATE finance.accounts
+						SET balance = balance + 1000
+						WHERE account_id IN (
+							SELECT account_id FROM public.users WHERE active = true
+					);`,
+			ExpectedObjects: []string{"finance.accounts", "public.users"},
+			ExpectedSchemas: []string{"finance", "public"},
+		},
+		{
+			Sql:             `SELECT classid, objid, refobjid FROM pg_depend WHERE refclassid = $1::regclass AND deptype = $2 ORDER BY 3`,
+			ExpectedObjects: []string{"pg_depend"},
+			ExpectedSchemas: []string{""},
+		},
+		{
+			Sql:             `SELECT pg_advisory_unlock_shared(100);`,
+			ExpectedObjects: []string{}, // empty slice represents no object collected
+			ExpectedSchemas: []string{},
+		},
+		{
+			Sql:             `SELECT xpath_exists('/employee/name', '<employee><name>John</name></employee>'::xml)`,
```

Comment:
can you add a test that has
-  pg_advisory_lock
-  xml_functions
- function call of another schema y
- table of schema x

Assert that objects returned are x.table, schemas returned are x.

Reply:
done